### DISCUSSION TITLE
A button next to the field form

### DISCUSSION THREAD

**Discussion by michalfita** [Budibase/budibase#5181] [Help]
My attempts end up with:
![image](https://user-images.githubusercontent.com/4925040/160875478-055049fc-6a15-4d7f-bd8e-e10da2742f7c.png)

I'd like a "+" button next to fields which refer to another table, so I can easily jump and add new row there then go back to editing or adding in the original form. How to do this?

Ideal solution would be if the button can display like a _subform_ to add, but I don't expect Budibase to be capable of doing so (for example I don't see how to build reusable forms to share between various screens.

**Comment by mjashanks** [comment]
Hi @michalfita - have you tried using a modal?

1. Create a new screen (your subform)
2. For the "Save" button, add an action in "Define Actions", to "Close Modal Screen"
3. Now, for the actions in your `+` button, add `Navigate To`, enter the URL of your new screen, and check the "Open in modal" box

**Comment by michalfita** [comment]
Thanks, @mjashanks that works really well. I still would prefer something what expands inside existing form, this should work for my cases.

But it answers only the second part of my question. I still struggle how to have the "+" button next to the text field of the row in the form on the right side.

![image](https://user-images.githubusercontent.com/4925040/160936264-3687ee84-0cb1-4ba5-b384-2103b37eb5bb.png)
That's more less what I'd like to achieve on the original form - I know how to make "+" button work, but now I need it to place it in the right location.

**Reply by dannyshin76** [reply]
Have you tried:
1. Move that field out of the field group
2. Create a ROW direction container, and create another container with STRETCH and a button into that container
3. Put the field in that nested container

Result:
![image](https://user-images.githubusercontent.com/93232582/160946606-7c370693-6270-48e1-9f94-630676bf1271.png)
![image](https://user-images.githubusercontent.com/93232582/160946625-875aead3-bb42-4648-8247-bb2d5cedc807.png)
